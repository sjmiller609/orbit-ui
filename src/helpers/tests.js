import { BrowserRouter } from 'react-router-dom'
import { shape } from 'prop-types'
import { mount } from 'enzyme'

// Instantiate router context
const router = {
  history: new BrowserRouter().history,
  route: {
    location: {},
    match: {},
  },
}

const createContext = () => ({
  context: { router },
  childContextTypes: { router: shape({}) },
})

// wrap with router context - to mount RR links correctly
export function mountWrap(node) {
  return mount(node, createContext())
}
