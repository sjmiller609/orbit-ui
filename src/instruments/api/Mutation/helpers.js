const has = (str, find) => ~str.toLowerCase().indexOf(find.toLowerCase())

export const formErrors = ({ error, errors }) => {
  if (!error) return

  let err = JSON.stringify(error).toLowerCase()
  return errors.find(e => {
    if (e.key && has(err, e.key)) return e
    else if (e.allKeys && e.allKeys.every(k => has(err, k))) return e
    else if (e.someKeys && e.someKeys.some(k => has(err, k))) return e
  })
}
