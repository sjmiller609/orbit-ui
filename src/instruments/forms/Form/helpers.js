// faltten and stringify nested keys
export const unpack = (obj, ns, hash) => {
  if (!obj) return
  return Object.entries(obj).reduce((acc, [key, value]) => {
    const keypath = ns ? `${ns}.${key}` : key
    if (value && typeof value === 'object') {
      return unpack(value, keypath, acc)
    }
    acc[keypath] = value
    return acc
  }, hash || {})
}

const roll = (keys, obj, value) => {
  const key = keys[0]
  // case 1 - not obj
  if (keys.length === 1) {
    obj[key] = value
    return obj
  }
  // case 2 - doesn't exist
  if (!obj[key]) {
    obj[key] = isNaN(keys[1]) ? {} : []
  }
  // case 3 - exists
  obj[key] = roll(
    keys.slice(1),
    Array.isArray(obj[key]) ? [...obj[key]] : { ...obj[key] },
    value
  )

  return obj
}

// convert back into obj
export const pack = (obj, type = {}) => {
  return Object.entries(obj).reduce((acc, [key, value]) => {
    const keys = key.split('.')

    return roll(keys, acc, value)
  }, type)
}

export const packChild = ({ name, obj }) => {
  const obj2 = {}
  let isArray
  Object.entries(obj).forEach(k => {
    if (name.length < k[0].length && k[0].indexOf(name) === 0) {
      const n = k[0].slice(name.length + 1)
      // must be true for all matching
      isArray = !isNaN(n.slice(0, n.indexOf('.')))
      obj2[n] = k[1]
    }
  })
  if (Object.keys(obj2).length) return pack(obj2, isArray ? [] : {})
  return null
}

export const removeChild = ({ name, obj = {} }) => {
  const obj2 = {}
  Object.entries(obj).forEach(k => {
    if (name.length >= k[0].length || k[0].indexOf(name) !== 0) {
      obj2[k[0]] = k[1]
    }
  })
  return obj2
}
