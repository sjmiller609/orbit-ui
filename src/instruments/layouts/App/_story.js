import React from 'react'
import { storiesOf } from '@storybook/react'
import { App } from 'instruments'

storiesOf('Instruments|Layouts.App', module)
  .add('Default', () => (
    <App nav={<div>Nav Element</div>} metaTitle="Mock Meta Title">
      [children]
    </App>
  ))
  .add('Full Height', () => (
    <App nav={<div>Nav Element</div>} metaTitle="Mock Meta Title" fullHeight>
      [children]
    </App>
  ))
