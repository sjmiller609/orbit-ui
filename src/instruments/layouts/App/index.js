'use strict'
import React from 'react'
import PropTypes from 'prop-types'

import classnames from 'classnames'

import s from './styles.scss'

import { Page } from 'instruments'

// Wrapper for all modules
//
// FullHeight is a way for child modules to signal that this wrapper should
// be 100% height. This was added to primarily allow the log view to be "full height",
// while allowing the other pages to behave normally, while also maintaining the bottom padding
// we add via the content class. If every module was "full height", the padding does not work.
// There may be a better way to do this.
class App extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      loading: true,
    }
  }

  componentWillMount() {
    setTimeout(this.handleLoading, 100)
  }

  handleLoading = () => {
    this.setState({ loading: false })
  }

  render() {
    const { children, nav, metaTitle, className, fullHeight } = this.props
    const { loading } = this.state
    return (
      <Page className={classnames(s.module, className)} metaTitle={metaTitle}>
        {nav}
        {!loading && (
          <div
            className={classnames(s.content, fullHeight ? s.fullHeight : '')}>
            {children}
          </div>
        )}
      </Page>
    )
  }
}

App.propTypes = {
  children: PropTypes.oneOfType([PropTypes.array, PropTypes.element])
    .isRequired,
  nav: PropTypes.element,
  metaTitle: PropTypes.string,
  className: PropTypes.string,
  fullHeight: PropTypes.bool,
}

export default App
