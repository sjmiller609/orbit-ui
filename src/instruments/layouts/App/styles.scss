@import 'src/styles/mixins.scss';

.content {
  position: relative;
  margin-top: -5.3rem;
  padding: 0 1% 15rem;
}

.fullHeight {
  height: 100%;
}
