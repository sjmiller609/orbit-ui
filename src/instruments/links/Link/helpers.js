// check if url is external
export const externalUrl = url => {
  if (typeof url === 'object') return false
  const path = /^https?:\/\/|^\/\//i
  if (path.test(url) && !~url.indexOf(window.location.origin)) return true
  return
}
