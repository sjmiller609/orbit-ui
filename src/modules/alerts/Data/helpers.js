import { formErrors } from 'instruments'

const errors = [
  {
    key: 'permissions',
    name: 'properties.alert_emails',
    error: 'Contact your workspace admin to upgrade your permissions.',
  },
]

export const handleError = error => formErrors({ error, errors })

export const trimError = error => error.split(':')[1].trim()
