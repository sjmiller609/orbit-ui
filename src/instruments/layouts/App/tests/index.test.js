import React from 'react'
import renderer from 'react-test-renderer'
import { mountWrap } from 'helpers/tests'
import App from '../index'

const props = {
  children: ['1', '2', '3'],
  nav: <p>Nav</p>,
  metaTitle: 'Test',
  className: 'testClassString',
  fullHeight: false,
}

const propsFullHeight = {
  ...props,
  fullHeight: true,
}

describe('App', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mountWrap(<App {...props} />)
  })

  it('should be defined', () => {
    expect(App).toBeDefined()
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should handle loading func', () => {
    wrapper = renderer.create(<App {...props} />)
    const inst = wrapper.getInstance()
    const handleLoading = jest.fn(() => inst.handleLoading())
    handleLoading()
    expect(handleLoading).toHaveBeenCalled()
  })

  it('should update loading state', () => {
    wrapper.setState({ loading: false })
    const content = wrapper.find('div').at(1)
    expect(content).toBeDefined()
  })

  it('should handle full height', () => {
    wrapper = mountWrap(<App {...propsFullHeight} />)
    wrapper.setState({ loading: false })
    const content = wrapper
      .find('div')
      .at(1)
      .hasClass('fullHeight')
    expect(content).toBeTruthy()
  })
})
