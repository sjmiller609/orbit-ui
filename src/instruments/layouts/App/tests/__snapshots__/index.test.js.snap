// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<App
  className="testClassString"
  fullHeight={false}
  metaTitle="Test"
  nav={
    <p>
      Nav
    </p>
  }
>
  <Page
    className="module testClassString"
    metaTitle="Test"
  >
    <div
      className="page module testClassString"
    >
      <p>
        Nav
      </p>
      <HelmetWrapper
        defer={true}
        encodeSpecialCharacters={true}
      >
        <SideEffect(NullComponent)
          defer={true}
          encodeSpecialCharacters={true}
          title="Test"
          titleAttributes={Object {}}
        >
          <NullComponent
            defer={true}
            encodeSpecialCharacters={true}
            title="Test"
            titleAttributes={Object {}}
          />
        </SideEffect(NullComponent)>
      </HelmetWrapper>
    </div>
  </Page>
</App>
`;
